client/src/App.tsx

```diff
@@ -18,6 +18,7 @@ export default function App() {
   const [user, setUser] = useState<User>();
   const [token, setToken] = useState<string>();
   const [isAuthorizing, setIsAuthorizing] = useState(true);
+  const navigate = useNavigate();
 
   // Load upcoming events for viewing
   useEffect(() => {
```

Comment:
On line 29, you'll want to display an error message to the user. You can just use `alert(msg)` for now. It's ugly and annoying, but it's easy and lets the user know something bad happened.